src/dune_rules/pkg_rules.ml

```diff
@@ -58,6 +63,22 @@ module Pkg_info = struct
   ;;
 end
 
+module Rule_root = struct
+  type dev_tool = Ocamlformat
+
+  type t =
+    | Project_deps
+    | Dev_tool of dev_tool
+
+  let to_string t =
+    match t with
+    | Project_deps -> ".pkg"
+    | Dev_tool Ocamlformat -> ".ocamlformat"
+  ;;
+
+  let equal t1 t2 = t1 = t2
```

Comment:
We should avoid structural equality and instead explicitly compare variants by matching.